### DISCUSSION TITLE
Batch Bioreactor Plot appearence issue

### DISCUSSION THREAD

**Discussion by taufeey** [do-mpc/do-mpc#345] [General]
![error in graph](https://user-images.githubusercontent.com/45310462/218299526-79b05498-b008-4466-8413-162bfd2cc37f.png)

I tried to run batch bioreactor code on the pycharm it executed upto 29 iteration but unfortunately the small figure window appears but does not show any results. please find the attachment and let me know what should i have to do.

Best Regards & Thanks.

**Comment by JoshuaAda** [comment]
Hi, 
a first suggestion from my side: Can you check which kind of backend your matplotlib is currently using? I noticed that especially the professional Pycharm version uses SciView in general for plotting and therefore uses a non-GUI backend like 'agg' . This means for example plt.show() would have no effect. Just take a look whether you can import tkinter (if not install it e.g on Ubuntu sudo apt-get install python3-tk)
and most importantly try matplotlib.use('TkAgg') to change the backend of matplotlib to a GUI-Backend. For reference this would be of help if this is indeed the problem:
https://stackoverflow.com/questions/48334853/using-pycharm-i-want-to-show-plot-extra-figure-windows (if you use Pycharm Pro and this plotting would be a problem connected with SciView)
https://stackoverflow.com/questions/56656777/userwarning-matplotlib-is-currently-using-agg-which-is-a-non-gui-backend-so
Hope it helps,
Joshua
